yaml/csr.yaml

```diff
@@ -1463,64 +1566,142 @@
       description: >
         Trap-handler base address, always aligned to 128 bytes
       type: WARL
-      reset_val: 0 # assumes mtvec_i == 0
+      reset_val: 0  # Note: assumes mtvec_i == 0
       msb: 31
       lsb: 7
     - field_name: BASE_6_2
       description: >
         Trap-handler base address, always aligned to 128 bytes
-      type: R
+      type: WARL
       reset_val: 0
       msb: 6
       lsb: 2
-    - field_name: MODE_1
+      warl_legalize: |
+        val_out = 0
+    - field_name: MODE
       description: >
-        CLIC mode enabled
+        0 = Direct mode, 1 = vectored mode
       type: WARL
-      reset_val: unknown
+      reset_val: 1
       msb: 1
-      lsb: 1
-    - field_name: MODE_0
+      lsb: 0
+      warl_legalize: |
+        val_out = val_in if (val_in == 0 or val_in == 1) else val_orig
+### ENDCOND
+
+### COND CLIC
+- csr: mtvec
+  description: >
+    Machine Trap-Vector Base Address
+  address: 0x305
+  privilege_mode: M
+  rv32:
+    - field_name: BASE_31_7
       description: >
-        0 = Direct mode, 1 = vectored mode
+        Trap-handler base address, always aligned to 128 bytes
       type: WARL
-      reset_val: 1
-      msb: 0
+      reset_val: 0  # Note: assumes mtvec_i == 0
+      msb: 31
+      lsb: 7
+    - field_name: BASE_6
+      description: >
+        Trap-handler base address, always aligned to 128 bytes
+      type: WARL
+      reset_val: 0
+      msb: 6
+      lsb: 6
+      warl_legalize: |
+        val_out = 0
+    - field_name: SUBMODE
+      description: >
+        Sub mode, reserved for future use
+      type: WARL
+      reset_val: 0
+      msb: 5
+      lsb: 2
+      warl_legalize: |
+        val_out = 0
+    - field_name: MODE
+      description: >
+        CLIC mode enabled
+      type: WARL
+      reset_val: 3
+      msb: 1
       lsb: 0
+      warl_legalize: |
+        val_out = 3
+### ENDCOND
+
+### COND CLIC
 - csr: mtvt
   description: >
     Machine Trap Vector Table Base Address
   address: 0x307
   privilege_mode: M
   rv32:
-    - field_name: BASE
+    - field_name: BASE_31_N
       description: >
         Trap-handler vector base address. Alignment depends on SMCLIC_ID_WIDTH parameter.
       type: WARL
       reset_val: 0
       msb: 31
+      lsb: 7
+    - field_name: BASE_N_1_6
+      description: >
+        Base address alignment field, hardcoded zero
+      type: WARL
+      reset_val: 0
+      msb: 6
+      lsb: 6
+      warl_legalize: |
+        val_out = 0
+    - field_name: Reserved_5_0
+      description: >
+        Reserved
+      type: R
+      reset_val: 0
+      msb: 5
       lsb: 0
+### ENDCOND
 
 - csr: mstatush
   description: >
     Machine ISA register
   address: 0x310
   privilege_mode: M
   rv32:
-    - field_name: SBE
+    - field_name: Reserved_31_6
```

Comment:
Sometimes, "reserved" bits are explicitly marked in this file, othertimes they are just omitted.